lib/sequent/core/command_service.rb

```diff
@@ -43,10 +43,22 @@ def remove_event_handler(clazz)
         event_store.remove_event_handler(clazz)
       end
 
+      def reset_lock_key
+        Thread.current[:sequent_command_service_lock_key] = :command_service_process_commands
+      end
+
+      def set_after_commit_lock
+        Thread.current[:sequent_command_service_lock_key] = :sequent_command_service_after_commit_lock_key
+      end
+
       private
 
+      def lock_key
+        Thread.current[:sequent_command_service_lock_key] ||= :command_service_process_commands
+      end
+
       def process_commands
-        Sequent::Util.skip_if_already_processing(:command_service_process_commands) do
+        Sequent::Util.skip_if_already_processing(lock_key) do
```

Comment:
Can't we better just "unset" the `command_service_process_commands` after the while block? This way we do not have to introduce new Thread local keys.

So `process_commands` will be:
```ruby
Sequent::Util.skip_if_already_processing(:command_service_process_commands) do
  begin                                                                       
    transaction_provider.transactional do                                     
      while(!command_queue.empty?) do                                         
        process_command(command_queue.pop)                                    
      end
      Sequent::Util.done_processing(:command_service_process_commands)                                            
    end                                                                       
  ensure                                                                      
    command_queue.clear                                                       
    repository.clear                                                          
  end                                                                         
end
```